project-base/storefront/components/Pages/ProductDetail/ProductDetailContent.tsx

```diff
@@ -40,9 +41,12 @@ export const ProductDetailContent: FC<ProductDetailContentProps> = ({ product, f
     const { isProductInComparison, toggleProductInComparison, isPopupCompareOpen, setIsPopupCompareOpen } =
         useComparison();
     const { toggleProductInWishlist, isProductInWishlist } = useWishlist();
+    const { setLastVisitedProductCatalogNumbers } = useLastVisited();
 
     useGtmProductDetailViewEvent(product, getUrlWithoutGetParameters(router.asPath), fetching);
 
+    setLastVisitedProductCatalogNumbers(product.catalogNumber);
+
```

Comment:
No way. You can't call function like this 🙈 You must use useEffect for this. Otherwise it will run with every rerender.

Reply:
Good point, I forgot do that.